tests/testcases/core/db_models/EEM_Term_Relationship_Test.php

```diff
@@ -60,6 +60,74 @@ public function test_insert_and_update_and_delete() {
 //		$this->assertEquals( 0, $tt_1->count() );
 //		$this->assertEquals(0,$tt_2->count() );
 	}
+
+
+    /**
+     * @group current
+     */
+	public function testUpdateTermTaxonomyCountsPassNull()
+    {
+        //create a few term taxonomies
+        $term1 = $this->factory()->term->create_and_get();
+        $term2 = $this->factory()->term->create_and_get();
+        $term3 = $this->factory()->term->create_and_get();
+        //add them to some posts
+        $post1 = $this->factory()->post->create_and_get();
+        $post2 = $this->factory()->post->create_and_get();
+        wp_set_post_terms($post1->ID, array($term1->term_id, $term2->term_id));
+        wp_set_post_terms($post2->ID,array($term2->term_id, $term3->term_id));
+        wp_cache_flush();
+        $term1 = get_term($term1->term_id);
+        $term2 = get_term($term2->term_id);
+        $term3 = get_term($term3->term_id);
+
+        $this->assertEquals(1, $term1->count);
+        $this->assertEquals(2, $term2->count);
+        $this->assertEquals(1, $term3->count);
+        //verify that when we update the counts, they aren't set to zero
+        EEM_Term_Relationship::instance()->update_term_taxonomy_counts();
+        wp_cache_flush();
+        $term1 = get_term($term1->term_id);
+        $term2 = get_term($term2->term_id);
+        $term3 = get_term($term3->term_id);
+        $this->assertEquals(1, $term1->count);
+        $this->assertEquals(2, $term2->count);
+        $this->assertEquals(1, $term3->count);
+    }
+
+
+    /**
+     * @group current
+     */
+    public function testUpdateTermTaxonomyCountsPassInt()
+    {
+        //create a few term taxonomies
+        $term1 = $this->factory()->term->create_and_get();
+        $term2 = $this->factory()->term->create_and_get();
+        $term3 = $this->factory()->term->create_and_get();
+        //add them to some posts
+        $post1 = $this->factory()->post->create_and_get();
+        $post2 = $this->factory()->post->create_and_get();
+        wp_set_post_terms($post1->ID, array($term1->term_id, $term2->term_id));
+        wp_set_post_terms($post2->ID,array($term2->term_id, $term3->term_id));
+        wp_cache_flush();
+        $term1 = get_term($term1->term_id);
+        $term2 = get_term($term2->term_id);
+        $term3 = get_term($term3->term_id);
+
+        $this->assertEquals(1, $term1->count);
+        $this->assertEquals(2, $term2->count);
+        $this->assertEquals(1, $term3->count);
+        //verify that when we update the counts, they aren't set to zero
+        EEM_Term_Relationship::instance()->update_term_taxonomy_counts($term2->term_id);
+        wp_cache_flush();
+        $term1 = get_term($term1->term_id);
+        $term2 = get_term($term2->term_id);
+        $term3 = get_term($term3->term_id);
+        $this->assertEquals(1, $term1->count);
+        $this->assertEquals(2, $term2->count);
+        $this->assertEquals(1, $term3->count);
+    }
```

Comment:
I don't get the purpose of having identical tests where the only difference is the name of the test.  Going by the name of the second test, I don't think its necessary to assert ints either because that's creating a test for WP behaviour right?  Either way, I don't think its necessary and maybe could just be removed?

Reply:
the setup and assertsions for both tests are the same, but the first one does `EEM_Term_Relationship::instance()->update_term_taxonomy_counts();` and the second does `EEM_Term_Relationship::instance()->update_term_taxonomy_counts($term2->term_id);`.
Before this ticket's work, the first one passed but the second one failed; with this work they both pass (last I checked!)

Reply:
does that explain why I did things this way? Do you still have recommendations?

Reply:
I probably should add some more info to each test's PHPdocs to explain what they're doing and what's different

Reply:
hmm ya I missed that.  I know I usually follow the rule of three before making code more DRY (and recommend that) but in this case it might be better to extract the boilerplate setup stuff into its own method here.  I think that may make the actual assertion differences stand out better if the same boilerplate isn't in the same tests.

Reply:
hmm ya good suggestion IMO